examples/infection-by-group/README.md

```diff
@@ -0,0 +1,380 @@
+* An `Infection Manager`, which watches for infections and schedules recoveries accordingly (based on the number of infected people in the region). Does so using rejection sampling because the rate of recovery is not known.
+* A `Region Manager`, which assigns people to regions (so there are `population_size/k` people in each region; panics if $population_size < k$), moves people between regions based on specified rates, and modifies the individual's recovery rate based on the number of infectious individuals in their region.
+* A `Report Writer` module, which listens for infections and recoveries and writes output to csv files, containing region-stratified disease patterns.
+
+Note that reading the input data will require some kind of parameter loading utility from `ixa` which reads from a config file, and exposes values to modules as global properties. This component must be mature enough to handle data that would be stored as a matrix or vector and put it into meaningful Rust data structures.
+
+As in other `ixa` models, the simulation is managed by a central `Context` object which loads parameters, initializes user-defined modules, and starts a callback execution loop:
+
+```rust
+struct Parameters {
+    random_seed: u64,
+    population_size: usize,
+    k: usize,
+    foi: HashMap<Counties, f64>,
+    base_infection_period: f64,
+    movement: HashMap<HashMap<Counties, f64>>
+}
+
+let context = Context::new();
+context::load_parameters<Parameters>("config.toml")
+
+// Initialize modules
+mod people
+mod transmission
+mod infection
+mod regions
+
+people.init();
+transmission.init();
+infection.init();
+regions.init();
+
+// Run the simulation
+context.execute();
+```
+
+### People and person properties
+
+People must be added to the simulation and have some properties assigned to them, such as their infection status. Their infection time is relevant for scheduling their recovery (more on this below). In this example, I tether assigning people to groups to moving them between regions, and do the assignment there. However, this is open to discussion.
+
+In `people.rs`:
+
+```rust
+enum Properties {infection_status, infection_time}
+// infection_time is Option<f64> with default value None
+enum InfectionStatus {S, I, R}
+
+fn init(context: &mut Context) {
+
+    // create the population and make them all susceptible -- ignoring their region for now
+    for id in 0..population_size {
+        let Person person = context.add_person();
+        person.set_property_value(Properties::infection_status, InfectionStatus::S);
+    }
+}
+```
+
+### Movement between regions
+
+In `regions.rs`:
+
+```rust
+use strum_macros::EnumIter;
+use ixa::GroupMappings::*;
+
+enum GroupTypes {Counties}
+
+#[derive(Debug, EnumIter)]
+enum Counties {A, B}
```

Comment:
Instead of defining `k` as a parameter, wouldn't it make more sense to have this derived from the total group types?